src/pages/Home.tsx

```diff
@@ -70,23 +73,36 @@ function Home() {
         }
         */
     }
-
-    // ensure useEffect is only called once
-    const [rendered, setRendered] = useState(false);
+    function renderQROrLoading() {
+        if (!isMobile()) {
+            return (
+                <>
+                    <TP>Scan the QR code with the Tonomy ID app</TP>
+                    {!showQR && <TProgressCircle />}
+                    {showQR && <QRCode value={showQR}></QRCode>}
+                </>
+            );
+        } else {
+            return (
+                <>
+                    <TP>Loading QR code request</TP>
+                    <TProgressCircle />
+                </>
+            );
+        }
+    }
 
     useEffect(() => {
-        setRendered(true);
-        if (!rendered) {
-            handleRequests();
-        }
-    }, [rendered]);
+        // console.log();
+        handleRequests();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
     return (
         <div style={styles.container}>
             <TImage height={62} src={require('../assets/tonomy/tonomy-logo1024.png')} alt="Tonomy Logo" />
             <TH1>{settings.config.appName}</TH1>
-            <TP>Loading QR code request</TP>
-            <TProgressCircle />
+            {renderQROrLoading()}
```

Comment:
this should show the loading component till it figures out that it is mobile or not

if mobile - show QR
if not mobile - redirect to app and after 1s (redirect to app fails) then redirect to DownloadApp page

Reply:
it is loading